examples/convert_ieeg_to_bids.py

```diff
@@ -43,92 +41,58 @@
-data = np.loadtxt(fname, dtype=str, delimiter='\t',
-                  comments=None, encoding='utf-8')
-column_names = data[0, :]
-info = data[1:, :]
-electrode_tsv = dict()
-for i, name in enumerate(column_names):
-    electrode_tsv[name] = info[:, i].tolist()
-
-# load in channel names
-ch_names = electrode_tsv['name']
-
-# load in the xyz coordinates as a float
-elec = np.empty(shape=(len(ch_names), 3))
-for ind, axis in enumerate(['x', 'y', 'z']):
-    elec[:, ind] = list(map(float, electrode_tsv[axis]))
-
-# %%
-# Now we make a montage stating that the iEEG contacts are in the MNI
-# coordinate system, which corresponds to the `fsaverage` subject in
-# FreeSurfer. For example, one can look at how MNE-Python deals with iEEG data
-# at `Working with SEEG
-# <https://mne.tools/stable/auto_tutorials/misc/plot_seeg.html>`_.
-montage = mne.channels.make_dig_montage(ch_pos=dict(zip(ch_names, elec)),
-                                        coord_frame='mni_tal')
-print(f'Created {len(ch_names)} channel positions')
-print(dict(zip(ch_names, elec)))
-
-# %%
-# We will load a :class:`mne.io.Raw` object and
-# use the montage we created.
-info = mne.create_info(ch_names, 1000., 'ecog')
-raw = mne.io.read_raw_edf(misc_path + '/ecog/sample_ecog.edf')
+raw = mne.io.read_raw_fif(op.join(
+    misc_path, 'seeg', 'sample_seeg_ieeg.fif'))
 raw.info['line_freq'] = 60  # specify power line frequency as required by BIDS
-raw.set_channel_types({ch: 'ecog' for ch in raw.ch_names})
-
-# set the bad channels
-raw.info['bads'].extend(['BTM1', 'BTM2', 'BTM3', 'BTM4', 'BTM5', 'BTM6',
-                         'BTP1', 'BTP2', 'BTP3', 'BTP4', 'BTP5', 'BTP6',
-                         'EKGL', 'EKGR'])
-
-# set montage
-# (note that this only works for some channel types: EEG/sEEG/ECoG/DBS/fNIRS)
-raw.set_montage(montage, on_missing='warn')
+subjects_dir = op.join(misc_path, 'seeg')  # Freesurfer recon-all directory
 
 # %%
-# Let us confirm what our channel coordinates look like.
-
-# make a plot of the sensors in 2D plane
-raw.plot_sensors(ch_type='ecog')
+# When the locations of the channels in this dataset were found
+# in `Locating Intracranial Electrode Contacts
+# <https://mne.tools/dev/auto_tutorials/clinical/10_ieeg_localize.html>`_,
+# the T1 was aligned to ACPC. So, this montage is in an
+# `ACPC-aligned coordinate system
+# <https://surfer.nmr.mgh.harvard.edu/fswiki/CoordinateSystems>`_.
+# We can either save the channel positions in the subject's anatomical
+# space (from their T1 image) or we can transform to a template space
+# such as ``fsaverage``. To save them in the individual space, it is
+# required that the T1 have been aligned to ACPC and then the channel positions
+# be in terms of that coordinate system. Automated alignment to ACPC has not
+# been implemented in MNE yet, so if the channel positions are not in
+# an ACPC-aligned coordinate system, using a template (like ``fsaverage``)
+# is the best option.
```

Comment:
In the email chain, it seems like people are advocating for using "other". If we can use "other" and set it to native/scanner RAS, then this is unambiguous and probably most widely understandable, as it's just related to `img.affine`, which is well understood and documented.

Reply:
-1 I don't see the merit in using coordinate system other in a tutorial about BIDS. I would rather just stick with `fsaverage` which at least is well-defined

Reply:
When BIDS changes to allow `scanner RAS`, I'm happy to update the tutorial but until then, I think it should not be recommended in a tutorial

Reply:
I think we should go back to BIDS with the proposal if it's not accepted at the moment. There are lots of holes in the standard and that's why tool developers need to complete the development loop. In the past, I and @sappelhoff had to iterate quite a bit between developing MNE-BIDS, realizing that there is an issue then fixing it in the standard + validator, then fixing the datasets and finally incorporating that back in MNE-BIDS. I am worried that we'll develop unnecessarily complicated pipelines involving too many coordinate transforms otherwise (you may remember how Tal complained when we presented the anonymization procedure to her which was overly complex)

Reply:
I hear you and I agree but I'm not sure how to get traction on the issue. It seems there is the MNE camp who thinks this is an issue and the everyone else camp who is happy to use "Other" and don't seem to be persuaded by the arguments in the discussion of this PR... not sure where to go from there. We've had @adam2392's issue up for a while, it has gone stale and we started an email thread with @agramfort which has also gone stale. I think the argument for scanner RAS is compelling but it hasn't been received that way so I'm not sure how to proceed in a different way to make that change

Reply:
@sappelhoff 's PR at https://github.com/bids-standard/bids-specification/pull/820 seems like it may address the concerns for having a "REQUIRED" pointer to the T1 image, if we choose to move forward with `scanner RAS` in BIDS, which encompasses the FreeSurfer T1.mgz, and possibly other reconstruction pipelines(?).

Reply:
honestly @alexrockhill I think you're doing a crazy gymnastic here but I don't think we should recommend. The data you use here is not ACPC aligned AFAIK so what @dorahermes says is that she would use "Other" and then refer to the T1.mgz origin coordinate system. That way simpler and not ambiguous (I think). Other is technically allowed by the standard and I prefer to add complexity to the code later than add it now and than possibly simplify later.

my 2c

Reply:
Not sure what you mean, this data is ACPC-aligned. I aligned the T1 before finding the coordinates.

Reply:
I think the only gymnastics are with the transformation to `fsaverage` with removing the fiducials. I agree that is not optimal but it seems like the only stable solution and *this PR isn't changing that, there were just no fiducials in the previous version so it wasn't explicit.*

Reply:
ok I had not realised these data were already ACPC aligned before running FS recon.

do you have time to discuss on discord?

Reply:
Part of my objection to how iEEG is handled in MNE is that fiducials need to be defined, which does not make a lot of sense when there is no need for a "head"-device transform and causes unnecessary confusion. See also [comment by @adam2392](https://github.com/mne-tools/mne-bids/pull/859#discussion_r693286578) here. I didn't quite follow the argument why [montage needed to be in "head" coordinates because BIDS required ACPC](https://github.com/mne-tools/mne-bids/pull/859#issuecomment-905824219).

Isn't it possible to store electrode positions in `scanras` if iEEG (or a different coordinate system if explicitly defined by user), require no fiducials and internally in MNE handle the visualization based on what coordinate system montage claims to be in? See [here](https://gist.github.com/jasmainak/a6ce60f1e957fe27a5ead32d7efb81e4) some code for my  project ... there is no need to define fiducials manually. Robert seems to have come around to scanras by the way.

Reply:
> Isn't it possible to store electrode positions in scanras if iEEG (or a different coordinate system if explicitly defined by user), require no fiducials and internally in MNE handle the visualization based on what coordinate system montage claims to be in

Up until the last few months, no, this was not possible with FIF. Now it is possible but many MNE functions would need to be changed to accommodate this. This discussion has come up many times now. I always make the same argument (and have already here): to me it's at worst a minor inconvenience to have to define this coordinate frame (especially with `estimate_mni_fiducials` or whatever it's called) and have to pass around `trans`, and it's the path of least resistance to make things work properly in MNE until we can refactor a bunch of viz and forward functions to handle EEG-like electrode positions that are not always in a properly defined head coordinate frame.

Reply:
> I didn't quite follow the argument why montage needed to be in "head" coordinates because BIDS required ACPC.

In that quoted issue, there were two separate things: what a "head" coordinate system is, and what an ACPC coordinate system is. I think they are separate issues, I don't think a montage needs to be in head because BIDS required ACPC.

Reply:
... it looks like @alexrockhill's previous answer to this question was in this PR in July for example:

https://github.com/mne-tools/mne-python/pull/9601/files#r675062176

> Yeah I agree it's not ideal but I think to be backward compatible with MEG/EEG, FIF requires it to be in the "head" frame. I've thought it a bit like how Freesurfer uses surface RAS as its internal representation frame, MNE uses "head". They are all arbitrary reference frames and at least there is a standard and you know how it's stored. Ideally the coordinate frame could be specified in the FIF but that seems like pretty large change.

@jasmainak let's not continue to discuss this here since I think of this as a "settled" issue (at least among @agramfort @adam2392 @alexrockhill ) even though it's not ideal, and mostly tangential to this PR. Feel free to open an MNE-Python issue where we can discuss further if you want, but please do search through previous discussions. Maybe it would help to have a Discord chat during office hours or something.

Reply:
sounds good. I understand that it requires a major refactor in MNE but seems like a necessary step to properly integrate sEEG + not bog down the user with coordinate systems. The issue of coordinate transforms I already mention in the [sEEG issue](https://github.com/mne-tools/mne-python/issues/8751#issue-786490766) that was raised some time ago. 

> it's too verbose about transformations. I think it's just an identity transformation usually.

I was clearly confused what was going on in the example. Do you want me to start another issue or we can keep that one open, perhaps converting the bullet points into check boxes? I think in general starting from the example and making sure it's simple enough is a good way to go.

Reply:
I can open an issue and I'd be happy to work on it when I have time, I think it might be already in the FIF specification so we might just have to change MNE